extension/lib/logger.js

```diff
@@ -108,24 +103,34 @@ Logger.prototype._writeAsync = function Logger_writeAsync(aMessage, aCallback) {
   // For testing purposes send the message to stdout
   //dump(aMessage + '\n');
 
-  // Create an output stream to write to file
-  var foStream = Cc["@mozilla.org/network/file-output-stream;1"]
-                 .createInstance(Ci.nsIFileOutputStream);
-  foStream.init(this._file, 0x02 | 0x08 | 0x10, PERMS_FILE, foStream.DEFER_OPEN);
-
-  // Write asynchronously to buffer;
-  // Input and output streams are closed after write
-  var iStream = this._converter.convertToInputStream(aMessage);
-  NetUtil.asyncCopy(iStream, foStream, function (status) {
-    if (!components.isSuccessCode(status)) {
-      var errorMessage = new Error("Error while writing to file: " + status);
-      console.error(errorMessage);
+  function callback(aResult) {
+    if (typeof(aCallback) === "function") {
+      aCallback(aResult);
     }
+  }
 
-    if (typeof(aCallback) === "function") {
-      aCallback(status);
+  let array = this._encoder.encode(aMessage);
+  let promise = OS.File.open(this.file.path, {write: true, append: true})
+  .then(file => {
+    file.write(array)
+    .then(bytes => {
+      file.close();
+      callback(bytes);
+    },
```

Comment:
Please do the formatting like https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/OSFile.jsm/OS.File_for_the_main_thread#Example.3A_Append_to_File. This is kinda hard to read right now.